chia/wallet/trading/offer.py

```diff
@@ -148,6 +150,27 @@ def __post_init__(self) -> None:
             if max_cost < 0:
                 raise ValidationError(Err.BLOCK_COST_EXCEEDS_MAX, "compute_additions for CoinSpend")
         object.__setattr__(self, "_additions", adds)
+        object.__setattr__(self, "_conditions", None)
+
+    def conditions(self) -> List[Condition]:
+        if self._conditions is None:
+            conditions: Dict[Coin, List[Condition]] = {}
+            max_cost = DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM
+            for cs in self._bundle.coin_spends:
+                try:
+                    cost, conds = cs.puzzle_reveal.run_with_cost(max_cost, cs.solution)
+                    max_cost -= cost
+                    conditions[cs.coin] = parse_conditions_non_consensus(conds.as_iter())
+                except Exception:  # pragma: no cover
+                    continue
+                if max_cost < 0:  # pragma: no cover
+                    raise ValidationError(Err.BLOCK_COST_EXCEEDS_MAX, "computing conditions for CoinSpend")
+            object.__setattr__(self, "_conditions", conditions)
```

Comment:
this looks like some trickery to fool `mypy`.. if `self._conditions` is an `Optional[]`, why not just use normal assignment?

Reply:
We don't necessarily need to do this computation for every `Offer` that we deal with so I'm basically making sure it only happens when we need it.  We set it to `None` during the `__post_init__` and only if we try to access `.conditions` do we do the computation.

Reply:
is it different from:
```
self._conditions = conditions
```
?

Reply:
Oh I see I thought you were talking about the spot in `__post_init__`.  I think we need to use this because the Offer class is a frozen dataclass.  This mechanism is how the `_additions` property etc. work as well.

Reply:
I see